multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1457,6 +1462,7 @@ def _top_k_average(
                             strict=strict_loading,
                         )
                         best_score = self.evaluate(val_df, [validation_metric_name])[validation_metric_name]
+                        self.best_score = best_score
```

Comment:
The self.best_score here is within the greedy soup, using which would cause error for uniform_soup or best.

Reply:
@zhiqiangdon Then where I can find the best validation score? For example, if using `best`, where is the val_accuracy located? Thank you.

Reply:
I think we can add
```self.best_score = self.evaluate(val_df, [validation_metric_name])[validation_metric_name]```
at line 1508 (https://github.com/awslabs/autogluon/pull/2316/files#diff-a73844176f6eae3b4538467b3565e3bdbf4d88997c3c670173bbb44e1a55b7a3R1508).

Reply:
Also, self.best_score hasn't considered matcher yet. Calling the fit_summary can cause errors for matcher.

Reply:
As discussed, add another line at 1508, self.best_score = self.evaluate(val_df, [validation_metric_name])[validation_metric_name]. 

But for the matcher issue, how should I address it? Any particular metric to get and where? Thank you.

Reply:
I think you need to add a similar line in matcher's top_k_average to compute best score and add a property in predictor to check which best_score (predictor's or matcher's) to use.

Reply:
I see. I think I will just do this for image classification in this PR, and leaves the rest for each feature owner.